dolphin/api/v1/volumes.py

```diff
@@ -11,20 +11,50 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import six
+from webob import exc
 
+from dolphin import db, exception
+from dolphin.api import api_utils
 from dolphin.api.common import wsgi
+from dolphin.api.views import volumes as volume_view
 
 
 class VolumeController(wsgi.Controller):
 
+    def __init__(self):
+        super(VolumeController, self).__init__()
+        self.search_options = ['name', 'status', 'id', 'storage_id', 'wwn', 'original_id', 'original_pool_id']
+
+    def _get_volumes_search_options(self):
+        """Return volumes search options allowed ."""
+        return self.search_options
+
     def index(self, req):
-        return dict(name="Storage volume 1")
+        ctxt = req.environ['dolphin.context']
+        query_params = {}
+        query_params.update(req.GET)
+        # update options  other than filters
+        sort_keys, sort_dirs = api_utils.get_sort_params(query_params)
+        marker, limit, offset = api_utils.get_pagination_params(query_params)
+        # strip out options except supported search  options
+        api_utils.remove_invalid_options(ctxt, query_params,
+                                         self._get_volumes_search_options())
+        try:
+            volumes = db.volume_get_all(ctxt, marker, limit, sort_keys, sort_dirs,
```

Comment:
Does volume_get_all throw any exception

Reply:
Only InvaliInput ,generic will be caught by framework

Reply:
Only InvaliInput ,generic will be caught by framework